cli/src/npm.ts

```diff
@@ -0,0 +1,135 @@
+import "zx/globals";
+import { z } from "zod";
+import { createHash } from "node:crypto";
+
+const repoRoot = path.resolve(__dirname, "../..");
+
+async function getTemplates(): Promise<string[]> {
+  return (await glob("./*-template/package.json")).map((t) => path.dirname(t));
+}
+
+export async function generateNpmLockfiles(): Promise<void> {
+  const config = await new TemplatesConfig().load();
+  const templates = await getTemplates();
+
+  for (const name of templates) {
+    echo(chalk.blue(`Updating template: ${chalk.grey(name)}`));
+    cd(path.resolve(repoRoot, name));
+
+    const packageJsonHash = await hashFile("./package.json");
+    const modified = config.updateTemplateHash(name, packageJsonHash);
+
+    if (!modified) {
+      echo(chalk.grey("package-lock.json already up to date, skipping"));
+      continue;
+    }
+
+    await fs.rm("./package-lock.json", { force: true });
+    await fs.rm("./node_modules", { force: true, recursive: true });
+
+    await $({
+      verbose: true,
+      stdio: "inherit",
+    })`npm install --no-audit --progress=false`;
```

Comment:
Think it might be better to using:

```sh
npm install --no-audit --progress=false --package-lock-only
```

This only generates the lockfile rather than actually installing deps

Reply:
TIL - good idea! If that ignores node_modules, we could stop removing node_modules and running pnpm install